IncludeToolbox/Formatter/IncludeFormatter.cs

```diff
@@ -47,16 +56,44 @@ public static string FormatPath(string absoluteIncludeFilename, FormatterOptions
         /// <summary>
         /// Formats the paths of a given list of include line info.
         /// </summary>
-        private static void FormatPaths(IEnumerable<IncludeLineInfo> lines, FormatterOptionsPage.PathMode pathformat, IEnumerable<string> includeDirectories)
+        private static void FormatPaths(IEnumerable<IncludeLineInfo> lines,
+                                        FormatterOptionsPage.PathMode pathformat,
+                                        FormatterOptionsPage.IgnoreFileRelativeMode ignoreFileRelativeMode,
+                                        IEnumerable<string> includeDirectories,
+                                        string documentDir,
+                                        string includeRootDirectory = null)
         {
             if (pathformat == FormatterOptionsPage.PathMode.Unchanged)
                 return;
 
             foreach (var line in lines)
             {
-                string absoluteIncludeDir = line.TryResolveInclude(includeDirectories, out bool resolvedPath);
+                string absoluteIncludePath = line.TryResolveInclude(includeDirectories, out bool resolvedPath);
                 if (resolvedPath)
-                    line.IncludeContent = FormatPath(absoluteIncludeDir, pathformat, includeDirectories) ?? line.IncludeContent;
+                {
+                    if (pathformat == FormatterOptionsPage.PathMode.Absolute_FromParentDirWithFile &&
+                        includeRootDirectory != null &&
+                        !absoluteIncludePath.StartsWith(includeRootDirectory))
+                    {
+                        continue;
+                    }
+
+                    var currentPathFormat = pathformat;
+                    var currentIncludeRootDirectory = includeRootDirectory;
+                    string includeFilename = Path.GetFileName(absoluteIncludePath);
+                    if ((ignoreFileRelativeMode == FormatterOptionsPage.IgnoreFileRelativeMode.InSameDirectory &&
+                         Path.Combine(documentDir, includeFilename) == absoluteIncludePath) ||
+                        (ignoreFileRelativeMode == FormatterOptionsPage.IgnoreFileRelativeMode.InSameOrSubDirectory &&
+                         absoluteIncludePath.StartsWith(documentDir)))
+                    {
+                        currentPathFormat = FormatterOptionsPage.PathMode.Absolute_FromParentDirWithFile;
+                        currentIncludeRootDirectory = documentDir;
```

Comment:
For these `IgnoreFileRelativeMode`s, we need to change the path format and root directory to root the include path appropriately.

Reply:
Not very happy with this. The combination of `FormatterOptionsPage.IgnoreFileRelativeMode.InSameDirectory`/`InSameDirectoryOrSubdirectory` with `FormatterOptionsPage.PathMode.Shortest` would no longer find the shortest possible path!
We need to pass on the `FormatterOptionsPage.IgnoreFileRelativeMode` to the inner FormatPath to be able to find the best path according to the `PathMode`

Reply:
I'll give that a try. I worry a bit that some options will be at cross-purposes with one another in some configurations (like "why would you do that?" configurations). If it makes things less awkward, though, I'm all for it!